test/ContentScript/integrationTests.test.ts

```diff
@@ -70,22 +71,98 @@ function integrationTests(
 
   if (browserName !== BROWSERNAME.FIREFOX) {
     test('Should Disable Extension', async () => {
+      server = getFakeZapServer(actualData, _JSONPORT);
       const context = await driver.getContext(_JSONPORT);
-      let page = await context.newPage();
-      await page.goto(await driver.getOptionsURL());
-      await page.uncheck('#zapenable');
-      await page.click('#save');
+      await driver.setEnable(false);
+      const page = await context.newPage();
+      await page.goto(
+        `http://localhost:${_HTTPPORT}/webpages/integrationTest.html`
+      );
+      await page.waitForLoadState('networkidle');
+      await page.waitForTimeout(1000);
       await page.close();
+      // Then
+      expect(JSON.stringify(Array.from(actualData))).toBe('[]');
+    });
 
-      page = await context.newPage();
+    test('Should record click', async () => {
+      // Given / When
+      server = getFakeZapServer(actualData, _JSONPORT);
+      const context = await driver.getContext(_JSONPORT, true);
+      await driver.setEnable(false);
+      const page = await context.newPage();
       await page.goto(
-        `http://localhost:${_HTTPPORT}/webpages/integrationTest.html`
+        `http://localhost:${_HTTPPORT}/webpages/interactions.html`
       );
+      await page.click('#click');
       await page.waitForLoadState('networkidle');
+      await page.waitForTimeout(1000);
+      await page.close();
+      // Then
+      expect(JSON.stringify(Array.from(actualData))).toBe(
```

Comment:
I think these would be much more readable if we created the objects and then stringified them.
Something like (untested;)
```
var expectedData = [{
  "action": {"action": "reportZestScript},
  ...
expect(JSON.stringify(Array.from(actualData))).toBe(JSON.stringify(Array.from(expectedData)))
```

which you've actually done later 😉

Reply:
okay

Reply:
Having a problem doing it this way because the data is being parsed twice once when it is sent to the fake zap server and once in the statement mentioned above